graphql/resolvers.go

```diff
@@ -2530,6 +2524,138 @@ func (r *queryResolver) BbGetCreatedTickets(ctx context.Context, taskID string)
 	return createdTickets, nil
 }
 
+// Will return an array of activated and unactivated versions
+func (r *queryResolver) MainlineCommits(ctx context.Context, options MainlineCommitsOptions) (*MainlineCommits, error) {
+
+	opts := model.MainlineCommitVersionOptions{
+		Activated:       true,
+		SkipOrderNumber: 0,
+		Limit:           0,
+	}
+	if options.Limit != nil {
+		opts.Limit = *options.Limit
+	}
+	if options.SkipOrderNumber != nil {
+		opts.SkipOrderNumber = *options.SkipOrderNumber
+	}
+
+	activatedVersions, err := model.GetMainlineCommitVersionsWithOptions(options.ProjectID, opts)
+	if err != nil {
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("Error getting activated versions %s", err.Error()))
+	}
+	opts = model.MainlineCommitVersionOptions{
+		Activated:       false,
+		SkipOrderNumber: 0,
+		Limit:           0,
+	}
+	if options.Limit != nil {
+		opts.Limit = *options.Limit
+	}
+	if options.SkipOrderNumber != nil {
+		opts.SkipOrderNumber = *options.SkipOrderNumber
+	}
+
+	unactivatedVersions, err := model.GetMainlineCommitVersionsWithOptions(options.ProjectID, opts)
+	if err != nil {
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("Error getting unactivated versions %s", err.Error()))
+	}
+	versions := append(activatedVersions, unactivatedVersions...)
+	sort.Slice(versions, func(i, j int) bool {
+		return versions[i].RevisionOrderNumber > versions[j].RevisionOrderNumber
+	})
+	var mainlineCommits MainlineCommits
+	activatedVersionCount := 0
+	for _, v := range versions {
+		if activatedVersionCount == *options.Limit {
+			break
+		}
+		apiVersion := restModel.APIVersion{}
+		err = apiVersion.BuildFromService(&v)
+		if err != nil {
+			return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error building APIVersion from service: %s", err.Error()))
+		}
+
+		// If we try to access a version that does not have the Activated flag we must manually verify it
```

Comment:
We would want to remove this block now right, since you're going to backfill activated?

Reply:
Ignore this, just saw your comment about not backfilling the field